### PR TITLE
Move repo certificate info generation to server side

### PR DESCRIPTION
**Pull request by jannfis**
This PR moves some of the logic generating repository certificate information from the client to the server side. 

It changes the following:

- Rename field `CertFingerprint` in `RepositoryCertificate` object of `RepositoryService` to `CertInfo`. This is a breaking change for the `RepositoryService` API, but since it is not released yet, now is still the chance to do it without much hassle. The name `CertInfo` far better suites the data contained, because the field can contain more than just a cert's fingerprint.
- Remove `CertData` from being set in `ListCertificates` operations. It's just data we don't need there, because it is never used anywhere. It will be sent in the soon-to-come `GetCertificate` operation of `CertificateService`
- Generate SHA256 fingerprint of the SSH key on the server side, not on the client. It will now be displayed in the Certificate overview of ArgoCD UI as well.
- Generate DN string for TLS cert subjects on the server side, not on the client. It will now be displayed in the Certificate overview of ArgoCD UI as well.

Cert List in Argo UI now looks like that:

image

### CODE REVIEW

**Comment by alexec** [comment]
@alexmt - I think we should try and get this into v1.2 because it becomes a breaking change if we don't.

**Comment by alexmt** [comment]
Agree. Now is a good time to change it

File: pkg/apis/application/v1alpha1/types.go
Comment (alexec) [inline pkg/apis/application/v1alpha1/types.go]: should the JSON names be Java case? I.e. `json:certFingerprint` rather than `json:certfingerprint`

File: util/cert/cert_test.go
Comment (alexec) [inline util/cert/cert_test.go]: minor - why check `err` here?

**Review by alexec — DISMISSED** [review]

**Review by alexmt — APPROVED** [review]
LGTM after addressing comments from @alexec .

File: util/cert/cert_test.go
Comment (jannfis) [reply util/cert/cert_test.go]: oops, good catch. Dunno how that came there, it definitely doesn't belong there

File: pkg/apis/application/v1alpha1/types.go
Comment (jannfis) [reply pkg/apis/application/v1alpha1/types.go]: You're right, they should be camel cased to improve readability. However, changing the names now would also require changing some code in the UI, because some of the variable names in the UI (in the parts interfacing with the API) are interdependent with the JSON names. So I see two options:

- Change the JSON names now and also bring in the required UI changes to this PR
- Have a separate PR shortly after this has been merged which changes case on the variable names in types.go and the UI parts. Personally, I think this option would be the better one.

Which one do you prefer?

Comment (alexec) [reply pkg/apis/application/v1alpha1/types.go]: lets do separate PR for the name change

**Pull request merged**
